<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>
  SPDX-FileCopyrightText: none
  SPDX-License-Identifier: GPL-3.0-or-later
 </author>
 <class>SSHTreeWidget</class>
 <widget class="QWidget" name="SSHTreeWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>506</width>
    <height>831</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Shortcut</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QKeySequenceEdit" name="keySequenceEdit"/>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLineEdit" name="filterText">
       <property name="placeholderText">
        <string>Filter...</string>
       </property>
       <property name="clearButtonEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btnInvertFilter">
       <property name="toolTip">
        <string>Show entries that don't match the filter</string>
       </property>
       <property name="text">
        <string>Invert</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_1">
     <item>
      <widget class="QCheckBox" name="btnManageProfile">
       <property name="toolTip">
        <string>Change profile on connect/disconnect</string>
       </property>
       <property name="text">
        <string>Manage profile</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="SshTreeView" name="treeView">
     <attribute name="headerVisible">
      <bool>false</bool>
     </attribute>
    </widget>
   </item>
   <item>
    <widget class="KMessageWidget" name="errorPanel"/>
   </item>
   <item>
    <widget class="QWidget" name="sshInfoPane" native="true">
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Identifier</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="name">
        <property name="toolTip">
         <string>A name for this ssh entry</string>
        </property>
        <property name="placeholderText">
         <string>The name of this entry on the list</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>SSH Hostname</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="hostname">
        <property name="toolTip">
         <string>The hostname of the server</string>
        </property>
        <property name="placeholderText">
         <string>The hostname of the server</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="port">
        <property name="toolTip">
         <string>The port of the server, 22 is the default for ssh connections</string>
        </property>
        <property name="text">
         <string>22</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>SSH Key</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLineEdit" name="sshkey">
          <property name="toolTip">
           <string>The private key to be used for this connection</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>The private key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnFindSshKey">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="folder-open"/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="useSshConfig">
        <property name="toolTip">
         <string>If this is set, we don't rely on the SSH key set above, but on the configuration on ~/.ssh/config</string>
        </property>
        <property name="text">
         <string>Use ~/.ssh/config for key</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Username</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="username">
        <property name="toolTip">
         <string>The username for the connection</string>
        </property>
        <property name="placeholderText">
         <string>Your username</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="folderLabel">
        <property name="text">
         <string>Folder</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QComboBox" name="folder">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>For organizational purposes, this entry will be inside of this folder.</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentText">
         <string>Default</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Profile</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="profile">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>The profile to be active when the ssh session starts</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="newSSHConfig">
       <property name="toolTip">
        <string>Add a new SSH Entry</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset theme="list-add"/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnDelete">
       <property name="toolTip">
        <string>Delete selected SSH entry</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset theme="edit-delete"/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnEdit">
       <property name="toolTip">
        <string>Edit selected entry</string>
       </property>
       <property name="text">
        <string>Edit</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnAdd">
       <property name="toolTip">
        <string>Add new SSH Configuration</string>
       </property>
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="toolTip">
        <string>Cancel Add or Edit</string>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header>kmessagewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SshTreeView</class>
   <extends>QTreeView</extends>
   <header>sshtreeview.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>treeView</tabstop>
  <tabstop>newSSHConfig</tabstop>
  <tabstop>name</tabstop>
  <tabstop>hostname</tabstop>
  <tabstop>port</tabstop>
  <tabstop>sshkey</tabstop>
  <tabstop>username</tabstop>
  <tabstop>btnAdd</tabstop>
  <tabstop>btnCancel</tabstop>
  <tabstop>btnEdit</tabstop>
  <tabstop>btnDelete</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
